internal/admin/users_handlers.go

```
@@ -0,0 +1,233 @@
+
+func (h *handler) guardLastAdminDemotion(r *http.Request, email string, newRole adminusers.Role) error {
+	if newRole == adminusers.RoleAdmin {
+		return nil
+	}
+	return h.guardLastAdminRemoval(r, email)
+}
+
+func (h *handler) guardLastAdminRemoval(r *http.Request, email string) error {
+	target, err := h.deps.UserStore.GetUser(r.Context(), email)
+	if err != nil {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 210) but it returns nil

(nilerr)